builder/builder.go

```diff
@@ -850,3 +874,72 @@ func parseUint32(s string) (uint32, error) {
 	}
 	return uint32(parsed), nil
 }
+
+// createCompressedArchive will use tar and xz to create a compressed
+// file. It does not stream the sources contents, doing all the work
+// in memory before writing the destination file.
+func createCompressedArchive(dst string, srcs ...string) error {
+	// Helper function to avoid repeating the destination
+	// information in every error we make.
+	wrapError := func(err error) error {
```

Comment:
This is still missing my point - sorry I have not been clearer.

If you want to write it like this then it should say something like

    func createCompressedArchive(dst string, srcs ...string) error {
        err := func (dst string, srcs ...string) error { // anonymous internal function
               var err error
               archive := &bytes.Buffer{}
               xw, err := swupd.NewExternalWriter(archive, xz)
               if err := nil {
                      return err   // just return err from the anonymous function
               }
               // lots of omitted lines
               return nil
            }(dst, srcs...)
         return errors.Wrapf(err, "couldn't create compressed archive %s", dst)
    }

Reply:
@icarus-sparry  Could you take another look?